inc/Engine/Media/Lazyload/Content/Processor/Regex.php

```diff
@@ -0,0 +1,51 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Media\Lazyload\Content\Processor;
+
+class Regex {
+	public function add_locations_hash_to_html( $html ) {
+		$result = preg_match( '/(?><body[^>]*>)(?>.*?<\/body>)/is', $html, $matches );
+
+		if ( ! $result ) {
+			return $html;
+		}
+
+		return $this->add_hash_to_element( $html, $matches[0], 2 );
+	}
+
+	private function add_hash_to_element( $html, $element, $depth ) {
+		if ( $depth < 0 ) {
+			return $html;
+		}
+
+		$skip_tags = [
+			'div',
+			'main',
+			'footer',
+			'section',
+			'article',
+			'header',
+		];
```

Comment:
Wouldn't be interesting to put this array in a private/protected properties ?
Or keep it like this, and apply a filter on it so we can add/remove elements from it ?